src/dynamics_tendencies.jl

```diff
@@ -0,0 +1,216 @@
+"""
+Compute the spectral tendency of the surface pressure logarithm
+"""
+function surface_pressure_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                   Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   l2::Int,                       # leapfrog index 2 (time step used for tendencies)
+                                   C::Constants{NF}
+                                   ) where {NF<:AbstractFloat}
+
+
+    @unpack pres_surf= Prog 
+    @unpack pres_surf_tend = Diag.Tendencies
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,dumc,px,py = Diag.miscvars
+    @unpack nlev, dhs = C 
+
+    #Initialise mean fields. 
+    u_mean[:,:] = 0.0
+    v_mean[:,:] = 0.0
+    d_mean[:,:] = 0.0 #d-mean is calculated here but actually used in vertical_velocity_tendency!()
+
+    #..and calculate values
+    for k in 1:nlev
+        u_mean += u_grid[:,:,k]*dhs[k] 
+        v_mean += v_grid[:,:,k]*dhs[k]
+        d_mean += div_grid[:,:,k]*dhs[k]
+    end
+
+    #Now use the mean fields
+    grad!(pres_surf[:,:,l2], dumc[:,:,2], dumc[:,:,3])
+    px = gridded(dumc[:,:,2]*3600, scale=true)
+    py = gridded(dumc[:,:,3]*3600, scale=true) #3600 factor from Paxton/Chantry. I think this is to correct for the underflow rescaling earlier
+
+    pres_surf_tend = spectral(-u_mean.*px - v_mean.*py)
+    pres_surf_tend[1,1] = Complex{NF}(zero)
+
+end
+
+"""
+Compute the spectral tendency of the "vertical" velocity
+"""
+function vertical_velocity_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                     Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                     C::Constants{NF}
+                                     ) where {NF<:AbstractFloat}
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,sigma_tend,sigma_m,puv,px,py = Diag.miscvars
+    @unpack nlev,dhs = C 
+
+    #Initialise
+    sigma_tend[:,:,:] = 0.0
```

Comment:
what is `sigma`? `σ=p/p0` (fraction of surface pressure) is used for the vertical levels, which I thought about renaming to `σ_levels` for more clearity (I don't think we use the term _level_ for anything but the vertical)

Reply:
Yeah this is a notation hangover from previous Speedy versions. `sigma_tend` denotes the vertical velocity in sigma coords. There is also a quantity `sigma_m` which I am not sure how to interpret. See e.g. L222,  223 of https://github.com/eapax/speedy/blob/stochastic_rounding/source/dyn_grtend.f90

This needs to be renamed. How about just `vertical_velocity`?